docs/blog/2020-01-23-why-typescript-chose-gatsby/index.md

```diff
@@ -0,0 +1,188 @@
+
+- I can still make custom React components like `Layout` above to encapsulate complexity,and TypeScript still has the props verified
+- JSX support with TypeScript is dreamy, and I love it
+- The 'runtime' code inside `useEffect` will be transpiled and verified to be correct by TypeScript
+- The 'runtime' code can re-use the same libraries as the server-side rendering, but have strong guarantees about only being optionally available
+
+By not using any of the React `setState`-ish APIs, I can guarantee there is no "runtime" React rendering happening on a user's browser either. This means the HTML in the built file is exactly what someone will see whether they have JS enabled or not. One advantage of this, has been that I can reliably run [`BackstopJS`](https://garris.github.io/BackstopJS/) to take screenshots of these static files to keep track of visual regressions as the site grows and others start to contribute.
+
+Would I recommend this constraint to people making Gatsby websites, probably not, it's going against the grain (React is a really good tool) of how you're expected to use Gatsby. But the trade-off is worth it for me, and it lowers the barrier if a compiler engineer wants to contribute.
+
+### Speed
+
+I'm blown away by how fast Gatsby is for a user.
+
+The founder of Gatsby, Kyle Mathews [gave a great talk in 2017 on the ways in which Gatsby is fast](https://youtu.be/Gtd-Ht-D0sg?t=961) and [here more recently](https://www.youtube.com/watch?v=HQEotVfTXwk), in rough:
+
+- Pre-fetching of related links
+- Clever splitting of code
+- Shrinking of assets
+- Offline support
+- Native lazy loading
+
+His long term visions is to think of Gatsby as a compiler which takes a set of input source files, and will keep trying to make a static output which is faster and faster for users. Another great resource for understanding the mechanics about why Gatsby is fast is this talk by [Nicolas Goutay at GOTO 2019](https://www.youtube.com/watch?v=p14g-Sep7HY).
+
+### Ecosystem Wins
+
+Gatsby has a massive community, and a lot of the time I'm able to just re-use someone elses work. The TypeScript team only has a few [unique](https://github.com/microsoft/TypeScript-Website/blob/v2/packages/gatsby-remark-shiki/README.md) [needs](https://github.com/microsoft/TypeScript-Website/tree/v2/packages/typescript-vfs) in [our code samples](http://www.typescriptlang.org/v2/dev/twoslash/) but other than that, it's a pretty vanilla website.
+
+It's been great to just add a single line into the [`gatsby-config.js`](https://github.com/microsoft/TypeScript-Website/blob/v2/packages/typescriptlang-org/gatsby-config.js), run `yarn add gatsby-plugin-something` and suddenly a whole new set of features are set up. Plugins tend to be small and easy to audit, I've used a few.
+
+### TypeScript Support
+
+While not "out-of-the-box", with a little elbow grease (as of early 2020) to get comprehensive TypeScript support:
```

Comment:
We should ensure that this matches our current docs for adding TS support.

Reply:
@ascorbic - he's right. We don't have official docs for adding typescript support yet. Let's run this as-is

Reply:
@hashimwarren Let's fix that by making TypeScript work out of the box

Reply:
But yes, not today!